specification/web/resource-manager/Microsoft.DomainRegistration/stable/2024-11-01/Domains.json

```diff
@@ -0,0 +1,1616 @@
+              "type": "boolean",
+              "readOnly": true
+            },
+            "managedHostNames": {
+              "description": "All hostnames derived from the domain and assigned to Azure resources.",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/HostName"
+              },
+              "readOnly": true,
+              "x-ms-identifiers": [
+                "name"
+              ]
+            },
+            "consent": {
+              "$ref": "#/definitions/DomainPurchaseConsent",
+              "description": "Legal agreement consent.",
+              "x-ms-mutability": [
+                "create"
+              ]
+            },
+            "domainNotRenewableReasons": {
+              "description": "Reasons why domain is not renewable.",
+              "type": "array",
+              "items": {
+                "enum": [
+                  "RegistrationStatusNotSupportedForRenewal",
+                  "ExpirationNotInRenewalTimeRange",
+                  "SubscriptionNotActive"
+                ],
+                "type": "string",
+                "x-ms-enum": {
+                  "name": "ResourceNotRenewableReason",
+                  "modelAsString": true
+                }
+              },
+              "readOnly": true
+            },
+            "dnsType": {
+              "description": "Current DNS type",
+              "enum": [
+                "AzureDns",
+                "DefaultDomainRegistrarDns"
+              ],
+              "type": "string",
+              "x-ms-enum": {
+                "name": "DnsType",
+                "modelAsString": false
+              }
+            },
+            "dnsZoneId": {
+              "description": "Azure DNS Zone to use",
+              "type": "string"
+            },
+            "targetDnsType": {
+              "description": "Target DNS type (would be used for migration)",
+              "enum": [
+                "AzureDns",
+                "DefaultDomainRegistrarDns"
+              ],
+              "type": "string",
+              "x-ms-enum": {
+                "name": "DnsType",
+                "modelAsString": false
+              }
+            },
+            "authCode": {
+              "type": "string",
+              "x-ms-mutability": [
+                "create",
+                "read"
+              ]
+            }
+          },
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "DomainAvailabilityCheckResult": {
+      "description": "Domain availability check result.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Name of the domain.",
+          "type": "string"
+        },
+        "available": {
```

Comment:
>available [](http://example.com/codeflow?start=9&length=9)

Prefer enum, e.g. availability: { CreateDomain, Default }, or other wording that fits better.